src/fidesctl/cli/utils.py

```diff
@@ -295,8 +290,48 @@ def handle_aws_credentials_options(
             credentials_config=fides_config.credentials,
             credentials_id=credentials_id,
         )
-        if not aws_config:
+        _validate_credentials_id_exists(credentials_id, aws_config)
+    return aws_config
+
+
+def handle_bigquery_config_options(
+    fides_config: FidesctlConfig,
+    dataset: str,
+    # keyfile_credentials: KeyfileCreds,
+    keyfile_path: str,
+    credentials_id: str,
+) -> Optional[BigQueryConfig]:
+    """
+    Handles the connections options for passing a keyfile, dictionary, or credentials-id.
+    """
+    bigquery_config = None
+    if keyfile_path:
+        if credentials_id:
             raise click.UsageError(
-                f"credentials-id {credentials_id} does not exist in fides config"
+                "Illegal usage: keyfile-path and credentials-id cannot be used together"
             )
-    return aws_config
+        with open(keyfile_path, "r", encoding="utf-8") as credential_file:
+            bigquery_config = BigQueryConfig(
+                **{
+                    "dataset": dataset,
+                    "keyfile_creds": json.load(credential_file),
```

Comment:
Feels a little odd that the CLI allows the credentials file as input but then just loads it into an object. It would make the most sense to me that if we get a `keyfile_path` we would just pass that into the client. We could choose from these options:

`generate dataset bigquery --private-key-id TOP_SECRET_ID --private-key TOP_SECRET ....`

`generate dataset bigquery --credentials-id`

`generate dataset bigquery --keyfile-path TOP_SECRET_FILE`

We don't have to implement them all though, for AWS we just allow explicit credentials, fides config credentials, or environment variable credentials

Reply:
Totally get what you mean here, my thinking was I could generate the engine in a consistent manner this way and avoid having to inspect and decide on a connection method based on what was passed (`boto3` infers this internally when connecting, right?), while also validating the keyfile itself. It's sort of a moot point however since we are inspecting everything as part of `handle_bigquery_config_options()` anyway... 

If we did have different engine connection options we would be able to avoid the creation of the temp file that had `fidesops` choose this pattern (another reason I went with it, consistency across repos) 

I'm down to give this another look and see what it looks like! I think there is also room for the env vars, so it would be nice to at least include all of the same options as AWS (potentially)